lib/OpenQA/WebAPI/Plugin/ObsRsync.pm

```diff
@@ -274,7 +308,13 @@ sub _get_run_last_info {
 
     my $linkpath = Mojo::File->new($home, $project, $batch, '.run_last');
     my $folder;
-    eval { $folder = readlink($linkpath) };
+    eval {
+        $folder = readlink($linkpath);
```

Comment:
In which case do you believe this `readlink` call will throw an exception?

Reply:
Let's assume I want to try the plugin on some exotic FS or let's in Windows. Other functionality will be still available, but because of this line - I couldn't try it at all. It is not question about thinking all possible scenarios - it is you attitude to what you want to have.

Reply:
Windows support is completely out of the question, we rely much too strongly on features Windows Perl could never support (i know that because many of those features are used in Mojolicious and Minion). And i would very much appreciate it if you would refrain from ad-hominem attacks in the future.

Reply:
It is not about support - not sure where it came from. It is about tolerance of edge case scenarios. It can even be some experimental custom build of perl or drivers or hardware etc - of course it is not supported, but still it can be useful in some scenarios. No need to pretend every possible edge case is known in advance. So why wouldn't just protect execution flow if particular failure is possible and can be tolerated (without regard how likely is the failure - "it is possible" and "it can be tolerated" is enough).

Reply:
Perhaps you should have taken a look at the [readlink](https://metacpan.org/pod/distribution/perl/pod/perlfunc.pod#readlink1) documentation first. It says exactly when it will throw an exception. The [reasons](https://metacpan.org/pod/distribution/perl/pod/perlport.pod#readlink) are very very narrow, and exclusively portability. The only justification for an `eval` here is support of more platforms. Non-standard builds of Perl are a total straw man argument.